include/margo-monitoring.h

```diff
@@ -0,0 +1,397 @@
+/**
+ * @file margo-monitoring.h
+ *
+ * (C) 2022 The University of Chicago
+ *
+ * See COPYRIGHT in top-level directory.
+ */
+#ifndef __MARGO_MONITORING_H
+#define __MARGO_MONITORING_H
+
+#include <abt.h>
+#include <mercury.h>
+#include <mercury_types.h>
+#include <mercury_bulk.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct hg_bulk_attr;
```

Comment:
I don't think the hg_bulk_attr forward ref is needed since the mercury headers are included here.

Reply:
I need it if the user has a mercury version < 2.2.0.

Reply:
Ah, Ok.  Maybe add a comment to note that.